src/threshold-ts/tbtc/index.ts

```diff
@@ -431,46 +439,121 @@ export class TBTC implements ITBTC {
         "Neither bitcoin client nor bitcoin credentials are specified"
       )
     }
-    this._bridge = new EthereumBridge({
-      address: Bridge.address,
-      signerOrProvider: getProviderOrSigner(
-        ethereumConfig.providerOrSigner as any,
-        ethereumConfig.account
-      ),
-    })
-    this._tbtcVault = getContract(
-      TBTCVault.address,
-      TBTCVault.abi,
-      ethereumConfig.providerOrSigner,
-      ethereumConfig.account
+    const {
+      chainId,
+      shouldUseGoerliDevelopmentContracts,
+      providerOrSigner,
+      account,
+    } = ethereumConfig
+
+    const tbtcVaultArtifact = getTbtcV2Artifact(
+      "TBTCVault",
+      chainId,
+      shouldUseGoerliDevelopmentContracts
+    )
+    const bridgeArtifact = getTbtcV2Artifact(
+      "Bridge",
+      chainId,
+      shouldUseGoerliDevelopmentContracts
     )
+    const tbtcTokenArtifact = getTbtcV2Artifact(
+      "TBTC",
+      chainId,
+      shouldUseGoerliDevelopmentContracts
+    )
+
     this._bridgeContract = getContract(
-      Bridge.address,
-      Bridge.abi,
-      ethereumConfig.providerOrSigner,
-      ethereumConfig.account
+      bridgeArtifact.address,
+      bridgeArtifact.abi,
+      providerOrSigner,
+      account
     )
+    this._tbtcVaultContract = getContract(
+      tbtcVaultArtifact.address,
+      tbtcVaultArtifact.abi,
+      providerOrSigner,
+      account
+    )
+    this._tokenContract = getContract(
+      tbtcTokenArtifact.address,
+      tbtcTokenArtifact.abi,
+      providerOrSigner,
+      account
+    )
+    // @ts-ignore
     this._bitcoinClient =
       bitcoinConfig.client ??
       new ElectrumClient(
         bitcoinConfig.credentials!,
         bitcoinConfig.clientOptions
       )
+
     this._multicall = multicall
+    this._ethereumConfig = ethereumConfig
     this._bitcoinConfig = bitcoinConfig
-    this._token = new EthereumTBTCToken({
-      address: TBTCToken.address,
-      signerOrProvider: getProviderOrSigner(
-        ethereumConfig.providerOrSigner as any,
-        ethereumConfig.account
-      ),
-    })
-    this._tokenContract = getContract(
-      TBTCToken.address,
-      TBTCToken.abi,
-      ethereumConfig.providerOrSigner,
-      ethereumConfig.account
-    )
+  }
+
+  async initializeSdk(
+    providerOrSigner: providers.Provider | Signer,
+    account?: string
+  ): Promise<SDK> {
+    const signer =
+      // Double bang to convert to boolean
+      !!account && providerOrSigner instanceof Web3Provider
+        ? getSigner(providerOrSigner as Web3Provider, account)
+        : providerOrSigner
+
+    const { shouldUseGoerliDevelopmentContracts } = this._ethereumConfig
+    const { client: clientFromConfig, credentials: credentialsFromConfig } =
+      this._bitcoinConfig
+
+    const hasMockedBitcoinClient = clientFromConfig && !credentialsFromConfig
```

Comment:
The bitcoin client should be mocked if we pass the client through the config regardless of whether we provide credentails or not. Having client in config means that the flag `MOCK_BITCOIN_CLIENT` is set to true, which means that the the client should be mocked adn we don't care if the user passes credentails or not.

If you want we can simplify it to:
```suggestion
    const hasMockedBitcoinClient = clientFromConfig 
```

But I would probably remove that variable and just use `clientFromConfig`.

Reply:
Ref commit: 23c8cb83